repl/src/main/scala/com/cloudera/livy/repl/Main.scala

```diff
@@ -135,17 +133,19 @@ class ScalatraBootstrap extends LifeCycle with Logging {
 
         // Wait for our url to be discovered.
         val replUrl = waitForReplUrl()
-
-        var req = url(callbackUrl).setContentType("application/json", "UTF-8")
-        req = req << write(Map("url" -> replUrl))
-
         info(s"Calling $callbackUrl...")
-        val rep = Http(req OK as.String)
-        rep.onFailure {
-          case _ => System.exit(1)
+        val response = Utils.usingResource(new AsyncHttpClient()) {
+          request => request.preparePost(callbackUrl)
+            .setHeader("Content-Type", "application/json;charset=UTF-8")
+            .setBody(write(Map("url" -> replUrl)))
+            .execute().get()
+        }
+        response.getStatusCode match {
```

Comment:
Is it safe to try to read the response's body after the client has been closed? (Not really familiar with the library's semantics.)